showcase/src/app/shared/example-viewer/example-viewer.component.ts

```diff
@@ -3,56 +3,92 @@ import { HttpErrorResponse } from "@angular/common/http";
 import { FileService } from "./file.service";
 import { STARK_LOGGING_SERVICE, StarkErrorImpl, StarkLoggingService } from "@nationalbankbelgium/stark-core";
 
+export interface ExampleFile {
+	extension: string;
+	data: string;
+	format: string;
+}
+
 @Component({
 	selector: "example-viewer",
 	templateUrl: "./example-viewer.component.html",
-	styleUrls: ["./example-viewer.component.scss"],
-	providers: [FileService]
+	styleUrls: ["./example-viewer.component.scss"]
 })
 export class ExampleViewerComponent implements OnInit {
 	@Input()
-	public extensions: string[] = ["HTML", "TS", "CSS"];
+	public extensions: string[];
 	@Input()
 	public filesPath: string;
 	@Input()
-	public title: string;
+	public exampleTitle: string;
 
-	public filesContent: object[] = [];
-	public showSource: boolean = false;
+	public appBaseHref: string;
+	public examplesFolder: string;
+	public exampleFiles: ExampleFile[];
+	public showSource: boolean;
 
-	public constructor(private service: FileService, @Inject(STARK_LOGGING_SERVICE) public loggingService: StarkLoggingService) {}
+	public constructor(private fileService: FileService, @Inject(STARK_LOGGING_SERVICE) public logger: StarkLoggingService) {
+		this.extensions = ["HTML", "TS", "CSS"];
+		this.exampleFiles = [];
+		this.showSource = false;
+		this.appBaseHref = this.getAppBaseHref();
+		this.examplesFolder = "assets/examples/";
+	}
 
 	public ngOnInit(): void {
-		this.fetchFiles();
+		this.fetchExampleFiles();
 	}
 
-	public fetchFiles(): void {
-		this.extensions.forEach((extension: string) => {
-			this.service.fetchFile("/assets/examples/" + this.filesPath + "." + extension.toLowerCase()).subscribe(
+	public fetchExampleFiles(): void {
+		for (const extension of this.extensions) {
```

Comment:
Is there a reason here to use for over forEach?

Reply:
Good question. Indeed, basically 2 reasons:

- Better performance: when transpiling TS code to Javascript, an `Array.forEach()` method is kept as is because is ES5 standard whereas `for...of` loop is converted to a basic `for` loop (when the target ES version is ES5) which is more performant in NodeJS and also in most Javascript engines of web browsers. I know that maybe in this specific case this loop will not have more than 3 or 5 iterations but I prefer to always use the `for...or` loop to keep consistency all over the code base and avoid confusions over when to use one syntax or the other.
- Shorter syntax: a `for...of` is shorter than a `forEach` because you don't need to specify the type of the item being iterated, it is inferred by Typescript.

Reply:
Allright I will also use for...of then :-)